### PR TITLE
Bower / Travis

### PR DESCRIPTION
**Pull request by GordonSmith**
Add bower support.
Add Travis-CI support.

Signed-off-by: Gordon Smith <gordon.smith@lexisnexis.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
@buunguyen Please Review
You can see the build report here:  https://travis-ci.org/GordonSmith/Visualization/builds/59778571
(I will need to tweak that so it works with upstream)

**Comment by GordonSmith** [comment]
@dineshshetye @jbrundage @mlzummo - You may want to review this as well...

**Comment by mlzummo** [comment]
Can you jump on IM?

File: bower.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    "dist/optimized/viz.min.css",
+    "dist/optimized/viz.common.min.js",
+    "dist/optimized/viz.other.min.js",
+    "dist/optimized/viz.tree.min.js",
+    "dist/optimized/viz.chart.min.js",
+    "dist/optimized/viz.graph.min.js",
+    "dist/optimized/viz.c3.min.js",
+    "dist/optimized/viz.google.min.js",
+    "dist/optimized/viz.map.min.js",
+    "dist/optimized/viz.layout.js",
```
Comment (buunguyen) [inline bower.json]: Should be viz.layout.min.js and viz.marshaller.min.js.

File: src/common/Widget.js [hunk 2]
```diff
@@ -3,6 +3,20 @@
     if (typeof define === "function" && define.amd) {
         define(["d3/d3"], factory);
     } else {
+        root.require = root.require || function (paths, cb) {
```
Comment (buunguyen) [inline src/common/Widget.js]: This shim doesn't handle word-cloud case yet. Do you want it fixed in this pull request or expect another pull request?

**Comment by mlzummo** [comment]
we/(I) need an updated readme so i can get this running from a fresh clone ... i have bower and npm installed

**Comment by mlzummo** [comment]
dermatology layer gives me an error on lodash

File: src/common/Widget.js (hunk 2 above)
Comment (GordonSmith) [inline src/common/Widget.js]: A separate PR will be fine - I don't have anyway to test it yet (until you get some non AMD demos going...).

**Comment by GordonSmith** [comment]
The demos also have their own dependencies - you need to:
```
cd demos
bower install
```

File: src/common/Widget.js (hunk 2 above)
Comment (buunguyen) [inline src/common/Widget.js]: I've just committed a full non-AMD demo and submitted a pull request to your repo. Funny thing I've found out is that we actually don't have to tweak the require shim at all, it would just work because the return value of d3-cloud module isn't used anywhere in the code.

**Comment by GordonSmith** [comment]
Fixes #269

**Comment by DeanPoulin** [comment]
:+1: Awesome!
